ini/parser.go

```diff
@@ -0,0 +1,317 @@
+				// check number of opened and closed sections []
+				if s.Count(line, "[") == 1 && s.Count(line, "]") == 1 {
+					parent = line[1 : len(line)-1]
+					p.addParent(parent)
+					newParent = true
+				} else {
+					err = InvalidSection
+				}
+
+				// parse sections values
+			} else if newParent && s.Count(line, "=") == 1 && (!contains([]string{"", "=", " "}, string(line[0])) && !contains([]string{"", "=", " "}, string(line[len(line)-1]))) {
+				if s.Contains(line, " = ") {
+					splitted := s.Split(line, " = ")
+					key = splitted[0]
+					val = splitted[1]
+				} else if s.Contains(line, "=") {
+					splitted := s.Split(line, "=")
+					key = splitted[0]
+					val = splitted[1]
+				}
+
+				p.add(parent, key, val)
+
+				// parse comment lines
+			} else if string(line[0]) == ";" {
+				continue
+
+				// invlid content
+			} else {
+				err = InvalidContent
+			}
+
+			// parse empty
+		} else if s.Trim(line, " ") == "" {
+			continue
+
+			// invlid content
+		} else {
+			err = InvalidContent
+		}
+	}
+
+	return err
+}
+
+func (p *Parser) ToDict() INIMap {
+	// Get the parsed dictionary as :
+	// [section1: map[key1: value1 key2: value2 ...] section2: map[key1: value1 key2: value2 ...] ...]
+
+	return p.parsedMap
+}
+
+func (p *Parser) GetSections() ([]string, error) {
+	// returns a list of all sections --> [section1 section2 ...]
+
+	var err error
+	var sections []string
+
+	for parent := range p.parsedMap {
+		sections = append(sections, parent)
+	}
+
+	// if no data in the parsed dict
+	if len(sections) == 0 {
+		err = NoParsedData
+	}
+
+	return sections, err
+}
+
+func (p *Parser) GetSection(sectionKey string) (Entry, error) {
+	// returns a dictionary for the section given --> map[key1: val1 key2: val2 ....]
+
+	sections, err := p.GetSections()
+	var section Entry
+
+	if err == nil && !contains(sections, sectionKey) {
+		err = SectionNotExist
+	} else {
+		section = p.parsedMap[sectionKey]
+	}
+	return section, err
+}
+
+func (p *Parser) GetOptions(sectionKey string) ([]string, error) {
+	// returns all options of the given section
+
+	section, err := p.GetSection(sectionKey)
+	var options []string
+
+	if err == nil {
+		for key := range section {
+			options = append(options, key)
+		}
+
+		if len(options) == 0 {
+			err = OptionsNotExist
+		}
+	}
+
+	return options, err
+}
+
+func (p *Parser) GetOption(sectionKey string, optionKey string) (string, error) {
+	// returns the value of the option key which belongs to the section key given
+
+	section, err := p.GetOptions(sectionKey)
+
+	var options []string
+	var option string
+
+	if err == nil {
+		for _, value := range section {
+			options = append(options, value)
+		}
+
+		if !contains(section, optionKey) {
+			err = OptionNotExist
+		} else {
+			option = p.parsedMap[sectionKey][optionKey]
+		}
+	}
+
+	return option, err
+}
+
+func (p *Parser) SetOption(sectionKey string, optionKey string, optionValue string) error {
```

Comment:
This should always work imo, it should create the section internally if it doesn't exist and set the key to the value, and the docs too please